File: pset6/caesar.py
import cs50
import sys

#make sure use only enetered one cmd line argument
if len(sys.argv) != 2:
    print("Enter only one whole number as key")
    exit(1)
    
#convert k user provided to int
k = sys.argv[1]
k = int(k)
k = k % 26

if type(k) is not int:
    print("please type an int key")
    exit(2)

#prompt user for plaintext
plaintext = cs50.get_string()
c = 0

#convert to ciphertext
for i in plaintext:
    if(i.islower is False):
        c = (((ord(i) + k - 65) % 26) +65)
        print("{}".format(chr(c)), end = "")
    elif(str.islower(i)):
        c = (((ord(i) + k - 97) % 26) +97)
        print("{}".format(chr(c)), end = "")
    else:
        print("{}".format(i), end = "")
        
        
print("")

File: pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main (int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Please enter one whole number as key\n");
        return 1;
    }
    
    //convert k user provided to int
    int k = atoi(argv[1]);
    k = k % 26;

    //prompt user for plaintext
    string plaintext = GetString();
    
    //display ciphertext
    for (int i = 0, n = strlen(plaintext); i < n; i++)
    {
        int c;
        
        if (isupper(plaintext[i]))
        {
            c = ((((int)plaintext[i] + k -65) % 26) + 65);
            printf("%c", c);
        }
        else if (islower(plaintext[i]))
        {
            c = ((((int)plaintext[i] + k - 97) % 26) + 97);
            printf("%c", c);
        }
        else
        {
            printf("%c", plaintext[i]);
        }
        
    }
    
    printf("\n");
    
    return 0;
    
}